### PR TITLE
Use tdr-github-actions workflows

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -3,42 +3,23 @@ on:
   push:
     branches:
       - master
-permissions:
-  id-token: write
-  contents: write
+      - test-reusable-workflows
```
Comment (TomJKing) [inline .github/workflows/build.yml:6]: Should this be left in here?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -3,42 +3,23 @@ on:
...
-        run: |
-          docker build -f Dockerfile-tests -t tests .
-          sbt assembly
-          docker build -t ${{ secrets.MANAGEMENT_ACCOUNT }}.dkr.ecr.eu-west-2.amazonaws.com/consignment-api:${{ env.NEW_VERSION }} .
-          docker push ${{ secrets.MANAGEMENT_ACCOUNT }}.dkr.ecr.eu-west-2.amazonaws.com/consignment-api:${{ env.NEW_VERSION }}
-          git tag ${{ env.NEW_VERSION }}
-          git push origin ${{ env.NEW_VERSION }}
-      - name: Deploy to integration
-        run: gh workflow run deploy.yml -f environment=intg -f toDeploy=${{ env.NEW_VERSION }}
+      - run: gh workflow run deploy.yml -f environment=intg -f to-deploy=${{ needs.pre-deploy.outputs.new-version }} --ref test-reusable-workflows
```
Comment (TomJKing) [inline .github/workflows/build.yml:23]: Should the branch `test-resusable-workflows` be referenced?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of minor comments.

File: .github/workflows/build.yml (hunk 1 above)
Comment (MancunianSam) [reply .github/workflows/build.yml:6]: Nope, thanks. that's just for testing, I've deleted it.

File: .github/workflows/build.yml (hunk 2 above)
Comment (MancunianSam) [reply .github/workflows/build.yml:23]: Same with this one, I've deleted it.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**